compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -147,6 +147,10 @@ impl<'a, 'tcx> TerminatorCodegenHelper<'tcx> {
 
     /// Call `fn_ptr` of `fn_abi` with the arguments `llargs`, the optional
     /// return destination `destination` and the unwind action `unwind`.
+    ///
+    /// You must provide `instance` if it might affect the alias set of the target.
+    /// This will never be the case for direct calls. It is currently only required
+    /// for `Virtual` calls, but providing it when available will avoid complications.
```

Comment:
I'm somewhat concerned about this requiring such "high-context" information. Most people have no idea what the hell this means by "alias set". Is there truly no other way to get the relevant Instance?

Reply:
The only value we have that is a per-instance value in this function right now is the `fn_ptr`, and I don't think the backends generally store instances for those because not all of them have one.

I do think your point about it being high-context whether an instance is necessary is reasonable though. Perhaps it would be better to describe it in terms of things you don't need an `Instance` for?  The reworded comment might read:

You must provide instance unless you are making a non-`Virtual` direct call or calling a raw `fn` pointer.

Reply:
sure.